EFCore.InMemory.HierarchyId/Extensions/InMemoryHierarchyIdDbContextOptionsBuilderExtensions.cs

```diff
@@ -0,0 +1,31 @@
+using System.Reflection;
+using Microsoft.EntityFrameworkCore.Infrastructure;
+using Microsoft.EntityFrameworkCore.InMemory.Infrastructure;
+
+namespace Microsoft.EntityFrameworkCore
+{
+    /// <summary>
+    /// HierarchyId specific extension methods for <see cref="InMemoryDbContextOptionsBuilder"/>.
+    /// </summary>
+    public static class InMemoryHierarchyIdDbContextOptionsBuilderExtensions
+    {
+        /// <summary>
+        /// Enable HierarchyId mappings.
+        /// </summary>
+        /// <param name="optionsBuilder">The builder being used to configure in-memory database.</param>
+        /// <returns>The options builder so that further configuration can be chained.</returns>
+        public static InMemoryDbContextOptionsBuilder UseHierarchyId(
+           this InMemoryDbContextOptionsBuilder optionsBuilder)
+        {
+            var optionsBuilderPropertyInfo = optionsBuilder.GetType().GetProperty("OptionsBuilder", BindingFlags.Instance | BindingFlags.NonPublic);
```

Comment:
reflection? 😨

Reply:
@smitpatel
I don't like this neither.

In Sql `.UseHierarchyId()` it was simpler, as the `SqlServerDbContextOptionsBuilder` is backed up by `IRelationalDbContextOptionsBuilderInfrastructure` which has the `OptionsBuilder` property.
But here, the `InMemoryDbContextOptionsBuilder` has no interface.

So I had to do this using reflection.

In order to avoid magic strings I wanted to use `nameof(InMemoryDbContextOptionsBuilder.OptionsBuilder)` but then I saw its property is `protected`.
And I didn't want to use other interfaces properties as property name reference for replacing the magic string.

Any other suggestion is appreciated.

Reply:
Filed https://github.com/dotnet/efcore/issues/23669